lang/py/src/avro/schema.py

```diff
@@ -551,6 +506,69 @@ def to_json(self, names=None):
   def __eq__(self, that):
     return self.props == that.props
 
+#
+# Decimal Fixed Type
+#
+
+class FixedDecimalSchema(FixedSchema, LogicalSchema):
+  def __init__(self, schema_type, size, name, precision, scale=0, namespace=None, names=None, other_props=None):
+    if schema_type != 'fixed':
+      raise SchemaParseException('Logical type DECIMAL must be backed by fixed or bytes.')
+    if size < 0:
+      raise SchemaParseException('Size of a fixed length should be greater than 0')
+
+    self._validate({'max_precision': self._max_precision(size),
+                   'precision': precision,
+                   'scale': scale,
+                   'size': size})
+
+    LogicalSchema.__init__(self, 'decimal')
+    FixedSchema.__init__(self, name, namespace, size, names, other_props)
+    self.set_prop('precision', precision)
+    self.set_prop('scale', scale)
+
+  def _validate(self, validation_props=None):
+    if not isinstance(validation_props.get('precision'), int):
+      raise SchemaParseException('Precision (Integer) is required for logical type DECIMAL.')
+
+    precision = validation_props.get('precision')
+    if precision <= 0:
+      raise SchemaParseException("Invalid DECIMAL precision %s. Must be positive."
+                                 % precision)
+    elif precision > validation_props.get('max_precision'):
+      raise SchemaParseException("Fixed %s cannot store precision digits. Max is %s"
+                                 %(validation_props.get('size'),
+                                   validation_props.get('max_precision')))
+
+    if not isinstance(validation_props.get('scale'), int):
+      raise SchemaParseException('Scale must be of type Integer.')
+
+    scale = validation_props.get('scale')
+    if scale < 0:
+      raise SchemaParseException("Invalid DECIMAL scale %s. Must be positive"
+                                 % scale)
+    elif scale > precision:
+      raise SchemaParseException("Invalid DECIMAL scale %s. Cannot be greater than precision %s"
+                                   %(scale, precision))
+
+  # read-only properties
+  precision = property(lambda self: self.get_prop('precision'))
+  scale = property(lambda self: self.get_prop('scale'))
+
+  def _max_precision(self, size):
+    return round(floor(log10(pow(2, (8 * size) - 1) - 1)))
+
+  def to_json(self, names=None):
+    new_props = self.props
+    new_props.update(self.other_props)
```

Comment:
It doesn't need to do this. The base `Schema` class already add `other_props` into `props.`